from launch import LaunchDescription
from launch_ros.actions import Node
from ament_index_python.packages import get_package_share_directory
import os

def generate_launch_description():
    # Path to the config file
    config_file_path = os.path.join(
        get_package_share_directory('task03'),
        'config',
        'task03.yaml'
    )

    return LaunchDescription([
        # Launch the trigger node with parameters from the config file
        Node(
            package='task03',
            executable='trigger_node',
            name='trigger_node',
            output='screen',
            parameters=[config_file_path],  # Load parameters from the config file
        ),
    ])
